oss-fuzz/arvo_reproducer.py

```diff
@@ -0,0 +1,1065 @@
+# ARVO reproducer
+# Paper: https://arxiv.org/abs/2408.02153
+# ARVO Implementation: https://github.com/n132/ARVO
+# Neil — May 5, 2025 — Seattle, USA
+# Jordi — July 30, 2025
+"""ARVO reproducer module.
+
+This module reproduces a vulnerability and its fix on OSS-Fuzz.
+Login gcloud:
+    $ gcloud auth application-default login 
+
+Classes:
+    BuildData: Named tuple for build configuration data.
+    
+Functions:
+    Main reproducing functions and utilities for OSS-Fuzz
+    vulnerability reproduction.
+"""
+
+import argparse
+import collections
+import json
+import logging
+import os
+import re
+import subprocess
+import tempfile
+import time
+from bisect import bisect_right
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Tuple, Union
```

Comment:
nit: use the more modern `| None` instead of optional, and `|` generally instead of `Union`.